### PR TITLE
CI should test with go's race detector

### PR DESCRIPTION
**Pull request by willscott**
cc @alanshaw

### CODE REVIEW

File: core/drand_test.go [hunk 1]
```diff
@@ -337,9 +337,9 @@ func (d *DrandTest2) StartDrand(id string, catchup bool, newGroup bool) {
 	dr := node.drand
 	// we load it from scratch as if the operator restarted its node
 	newDrand, err := LoadDrand(dr.store, dr.opts)
+	//newDrand.opts.clock = node.clock
```
Comment (alanshaw) [inline core/drand_test.go:340]: This is commented now were the tests failing?

Comment (willscott) [reply core/drand_test.go:340]: yep. it causes a race between a routine reading the clock that spins out from `LoadDrand`. it seems not needed, so :shrug:

Comment (alanshaw) [reply core/drand_test.go:340]: Gotcha, so, I'd either remove this line if it's really not needed, or if you don't know and don't have the time to look into it then open an issue to get back to fixing this _and_ add a comment here explaining why it's commented out.

Comment (willscott) [reply core/drand_test.go]: pushed to remove the line fully

Comment (nikkolasg) [reply core/drand_test.go]: That's correct - I can confirm it's not needed - the `dr.opts` comes from the previous node that was here before, so the clock is the same so there is no need to set it explicitely.

**Pull request merged**